### ISSUE TITLE
`digmiro` is not respected when merge target is not a Sierra record

### ISSUE THREAD

**Issue by jamieparkinson** [wellcomecollection/platform#5713]
As documented in https://wellcome.slack.com/archives/C8X9YKM5X/p1690457327387449 and the linked thread, we encounter a bug when we merge the following:
- A Sierra record with the `digmiro` digcode 
- MIRO images
- A CALM *or* TEI record for the Sierra record
- A METS record (which will be for the re-ingested MIRO images)

The intention of `digmiro` is to indicate that linked MIRO images have been reingested (and therefore have a METS record), meaning that the merger knows not to duplicate the corresponding images and items. However, this logic was implemented with the faulty assumption that the "target" would always be the Sierra record: this isn't true in the case that there is a CALM or a TEI record present, and we should work out how to handle that case.

**Comment by pollecuttn** [comment]
It would be good if whoever picks this up works on this with someone who doesn't know about this area so the knowledge is spread around the team.

**Comment by pollecuttn** [comment]
This also came up via https://wellcome.slack.com/archives/C8X9YKM5X/p1746534321906459

**Comment by pollecuttn** [comment]
Another example with a bit of context: https://wellcome.slack.com/archives/C8X9YKM5X/p1747302514306659

**Comment by kenoir** [comment]
See also: https://wellcome.slack.com/archives/C8X9YKM5X/p1747650108449699

**Comment by paul-butcher** [comment]
A query to find works relating to digmiro images where the final Work is not from Sierra:
```
GET works-indexed-2025-05-01/_search 
{
  "query": {
    "bool":{
      "must": [
        {"match":{"query.identifiers.value": "digmiro"}}
      ],
      "must_not": [
        
        {"regexp": {
          "query.sourceIdentifier.value": "b.*"
        }}
      ]
    }
  },
    "_source":["query.sourceIdentifier.value", "query.identifiers.value"]
}
```

**Comment by paul-butcher** [comment]
A better query - this one just show the 11 records that have digmiro and images
```

GET works-indexed-2025-05-01/_search?size=100
{
  "query": {
    "bool":{
      "must": [
        {"match":{"query.identifiers.value": "digmiro"}},
        {"exists": {"field":"query.images.id"}}
      ],
      "must_not": [
        
        {"regexp": {
          "query.sourceIdentifier.value": "b.*"
        }}
      ]
    }
  },
    "_source":["query.sourceIdentifier.value", "query.identifiers.value"]
}
```

Revealing that https://wellcomecollection.org/works/wck8hfaa is a problematic one.

These are all the non-sierra Works with digmiro and images:
```
r99y3a5y
wck8hfaa
dxy6f5df
a8q8dc6u
ryyvsjh5
e2579kmt
vrdjb5pq
qqxq3sga
bjyw7p84
vwq9spk2
j9hnuqw5
```

**Comment by paul-butcher** [comment]
This is a surprisingly niche problem. Out of 5578 non-sierra digmiro works, only 11 appear to be ignoring the digmiro suppression.

I'm not sure the diagnosis fits the problem. [annkhcve](https://wellcomecollection.org/works/annkhcve) (not on the above list)  seems to work as expected.  It's a CALM-derived Work associated with a digmiro-marked Sierra work (works-identified-2025-05-01/_doc/ppjprs6s)

![Image](https://github.com/user-attachments/assets/25e22580-3e1a-43e0-8ceb-e518cbe7084e)

wck8hfaa (on the above list) OTOH, is also a CALM-derived Work associated with a digmiro-marked Sierra work, and is emitting Images.  I wonder if the pertinent difference is that there are two Sierras?  (more investigation needed)

<img width="1868" alt="Image" src="https://github.com/user-attachments/assets/59604820-c5ab-4b8b-ab6a-1530e6c768cf" />

**Comment by paul-butcher** [comment]
Yes!  I believe this is it. (Just need to properly reproduce it in a minimal form to be sure)

For the Image fields, the actual target is ignored, and the fields are all merged as though the Sierra Work is the target.  The values then get applied to the proper target.

Part of the reason for this is that Sierra is where digmiro is defined, so we look at the Sierra-derived participant in the merge in order to work out whether to discard the images.

However, that assumes that there is one Sierra record participating in the merge, and that record is the one that has digmiro defined.

When there is more than one Sierra record, this bit of the merge logic just grabs the first one to hand, and pretends it's the target.  If that is not the one with digmiro defined, then the images are emitted.  

In the above example, ga6prd9b defines digmiro, and bv3wtb3h does not.

I surmise that bv3wtb3h is being chosen as the pretend target, and therefore the merge is behaving as though there is no digmiro.

**Comment by paul-butcher** [comment]
those records have now been reprocessed and look correct
